pkg/model/nodeinfo.go

```diff
@@ -6,10 +6,11 @@ import (
 	"github.com/libp2p/go-libp2p/core/peer"
 )
 
+//go:generate stringer -type=NodeType -trimprefix=NodeType -output=nodeinfo_string.go
 type NodeType int
 
 const (
-	nodeTypeUnknown NodeType = iota
+	NodeTypeRequester NodeType = iota
 	NodeTypeCompute
 )
```

Comment:
This will mean previous versions of the code treat unknown nodes as Requester nodes. Could we have instead made this a new value instead of modifying the existing?

Reply:
My careful reading of the code concluded that this was always true (i.e. the only time we used unknown is when we definitely knew it was a requester anyway) so it's not introducing any ambiguity here. If we add a new node type in the future, previous versions will continue to "not know" what type it is as long as we use an appropriate value.